backend/editor/github_functions.py

```diff
@@ -47,19 +59,28 @@ def update_file(self, filename):
         # Find taxonomy text file to be updated
         github_filepath = f"taxonomies/{self.taxonomy_name}.txt"
         commit_message = f"Update {self.taxonomy_name}.txt"
+        try:
+            current_file = self.repo.get_contents(github_filepath)
+            with open(filename, "r") as f:
+                new_file_contents = f.read()
 
-        current_file = self.repo.get_contents(github_filepath)
-        with open(filename, "r") as f:
-            new_file_contents = f.read()
-
-        # Update the file
-        self.repo.update_file(
-            github_filepath,
-            commit_message,
-            new_file_contents,
-            current_file.sha,
-            branch=self.branch_name,
-        )
+            # Update the file
+            self.repo.update_file(
+                github_filepath,
+                commit_message,
+                new_file_contents,
+                current_file.sha,
+                branch=self.branch_name,
+            )
+        except GithubException as e:
+            # Handle GitHub API-related exceptions
+            raise Exception(f"GitHub API error: {e}")
+        except FileNotFoundError:
+            # Handle file not found error (e.g., when 'filename' does not exist)
+            raise Exception(f"File not found: {filename}")
+        except Exception as e:
+            # Handle any other unexpected exceptions
+            raise Exception(f"An error occurred: {e}")
```

Comment:
nit: To keep the exception context

```suggestion
        except GithubException as e:
            # Handle GitHub API-related exceptions
            raise Exception(f"GitHub API error: {e}") from e
        except FileNotFoundError:
            # Handle file not found error (e.g., when 'filename' does not exist)
            raise Exception(f"File not found: {filename}") from e
        except Exception as e:
            # Handle any other unexpected exceptions
            raise Exception(f"An error occurred: {e}") from e
```

Reply:
good catch.